src/validation_workflow/download_utils.py

```diff
@@ -0,0 +1,29 @@
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+
+import requests
+
+from system.temporary_directory import TemporaryDirectory
+
+
+class DownloadUtils:
+    @staticmethod
+    def is_url_valid(url: str) -> int:
+        response = requests.head(url)
+        if response.status_code == 200:
+            return 1
+        else:
+            return 0
+
+    @staticmethod
+    def download(url: str, tmp_dir: TemporaryDirectory) -> bool:
+        # This method writes the contents from the response object into temporary directory file name fetched from the end of the url.
+        response = requests.get(url, stream=True)
+        path = tmp_dir.name + "/" + url.split("/")[-1]
```

Comment:
This is not a good implementation and I would rather use os.join because `/` is only working on nix* and would cause trouble on windows (even tho windows now can work with `/` it is still not ideal here.)

Reply:
Also, if you want to get the file name, use something like os.basename or similar, do not try to split the url and get the last item.